### PR TITLE
Add dynamodb_table module

### PR DESCRIPTION
**Pull request by loia**
- New module for AWS Dynamo DB
- Can be used to create, update and delete Dynamo DB tables.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks for submitting this module to Ansible Extras. Apologies that it’s taken a while to get your module reviewed.

To help facilitate reviews, we’ve broadened the number of people who can approve modules for inclusion into the Extras repository. The list of official reviewers can be found here:

https://github.com/ansible/ansible-modules-extras/blob/devel/REVIEWERS.md 

Our new policy is that if a new module is reviewed and approved by at least two official module reviewers, the module will be approved for inclusion.  We will be asking the community of reviewers to take a look at these modules on a regular basis.

To ensure that your module has the best chance of being approved, please double-check that you adhere to the following Ansible module guidelines: 

General guidelines: http://docs.ansible.com/developing_modules.html#module-checklist
AWS-specific guidelines: https://github.com/ansible/ansible-modules-extras/blob/devel/cloud/amazon/GUIDELINES.md

**Comment by loia** [comment]
Thanks Greg.  I've made some improvements after reviewing the module guidelines.

**Comment by gregdek** [comment]
Thanks @loia.

Pinging AWS reviewers: if any of you can review, please do. 

@jsmartin @scicoin-project @tombamford @garethr @lorin @jarv @jsdalton @silviud @adq @zbal @zeekin @willthames @lwade @carsongee @defionscode @tastychutney @bpennypacker

File: cloud/amazon/dynamodb_table.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True)
+
+    if not HAS_BOTO:
+        module.fail_json(msg='boto required for this module')
+
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module)
+    connection = boto.dynamodb2.connect_to_region(region)
```
Comment (willthames) [inline cloud/amazon/dynamodb_table.py]: Should be

```
connection = connect_to_aws(boto.dynamodb2, region, **aws_connect_params)
```

Comment (loia) [inline cloud/amazon/dynamodb_table.py]: @willthames Thanks for the review - I've applied your suggested change.

**Comment by willthames** [comment]
:+1: all looks good to me.

**Comment by gregdek** [comment]
Thanks @willthames! That's one review down. One more gets this module included. 

Re-pinging: @jsmartin @scicoin-project @tombamford @garethr @lorin @jarv @jsdalton @silviud @adq @zbal @zeekin @lwade @carsongee @defionscode @tastychutney @bpennypacker

**Comment by bpennypacker** [comment]
+1 - looks good

**Comment by gregdek** [comment]
Thanks @loia! This also makes you an approved reviewer, so be sure to pay it forward if you can. :)

File: cloud/amazon/dynamodb_table.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = """
+---
+module: dynamodb_table
+short_description: Create, update or delete AWS Dynamo DB tables.
```
Comment (bcoca) [inline cloud/amazon/dynamodb_table.py:20]: missing version_added

File: cloud/amazon/dynamodb_table.py [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+    required: false
+  hash_key_type:
+    description:
+      - Type of the hash key.
+    required: false
+    choices: ['STRING', 'NUMBER', 'BINARY']
+    default: 'STRING'
+  range_key_name:
+    description:
+      - Name of the range key.
```
Comment (bcoca) [inline cloud/amazon/dynamodb_table.py:52]: default?

**Comment by loia** [comment]
Submitted fixes in new pull request https://github.com/ansible/ansible-modules-extras/pull/654

**Pull request merged**